Source/SwiftLintFramework/Rules/ModifiersOrderRule.swift

```diff
@@ -0,0 +1,156 @@
+    public static let description = RuleDescription(
+        identifier: "modifiers_order",
+        name: "Modifiers Order",
+        description: "Modifiers order should be consistent.", kind: RuleKind.style,
+        nonTriggeringExamples: [
+            "@objc \npublic final class MyClass: NSObject {\n" +
+            "private final func myFinal() {}\n" +
+            "weak var myWeak: NSString? = nil\n" +
+            "public static let nnumber = 3 \n }",
+
+            "public final class MyClass {}"
+        ],
+        triggeringExamples: [
+            "@objc \npublic final class MyClass: NSObject {\n" +
+            "final private func myFinal() {}\n}",
+
+            "@objc \nfinal public class MyClass: NSObject {}\n",
+
+            "final public class MyClass {}\n",
+
+            "class MyClass {" +
+            "weak internal var myWeak: NSString? = nil\n}",
+
+            "class MyClass {" +
+            "static public let nnumber = 3 \n }"
+        ]
+    )
+
+    private func extractContinuousSections(from tokens: [SyntaxToken], in file: File) -> [[SyntaxToken]] {
+        var builtInTokenSections = [[SyntaxToken]]()
+        var tokenSection = [SyntaxToken]()
+        let builtIn = "source.lang.swift.syntaxtype.attribute.builtin"
+        let keyword = "source.lang.swift.syntaxtype.keyword"
+        let declClass = "source.lang.swift.decl.class"
+
+        for token in tokens {
+            // static & class are not builtin attibutes so we check every keyword for static || class
+            if token.type == keyword,
+                token.length == "static".bridge().length || token.length == "class".bridge().length,
+                let keywordAtLoc = file.contents.bridge()
+                                .substringWithByteRange(start: token.offset, length: token.length) {
+                    switch keywordAtLoc {
+                    case "static":
+                        tokenSection.append(token)
+                    // filter out `class` when used as a declaration
+                    case "class" :
+                        if let (kind, _) = file.structure.kinds(forByteOffset: token.offset)
+                                            .first(where: { $0.byteRange.location == token.offset }),
+                            kind != declClass {
+                                tokenSection.append(token)
+                            }
+                    default: continue
+                    }
+            } else if token.type == builtIn {
+                tokenSection.append(token)
+            } else if !tokenSection.isEmpty {
+                builtInTokenSections.append(tokenSection)
+                tokenSection.removeAll()
+            }
+        }
+        if !tokenSection.isEmpty {
+            builtInTokenSections.append(tokenSection)
+        }
+        return builtInTokenSections.filter({ $0.count > 1 })
+    }
+
+    private func isAccessControlLabel(_ input: String) -> Bool {
```

Comment:
another part that may be fragile. I could not find the facilities to get this information from the source kit.